make/_shared/tools/00_mod.mk

```diff
@@ -0,0 +1,490 @@
+# Copyright 2023 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ifndef bin_dir
+$(error bin_dir is not set)
+endif
+
+##########################################
+
+$(bin_dir)/scratch/image $(bin_dir)/tools $(bin_dir)/downloaded $(bin_dir)/downloaded/tools:
+	@mkdir -p $@
+
+checkhash_script := $(dir $(lastword $(MAKEFILE_LIST)))/util/checkhash.sh
+
+# To make sure we use the right version of each tool, we put symlink in
+# $(bin_dir)/tools, and the actual binaries are in $(bin_dir)/downloaded. When bumping
+# the version of the tools, this symlink gets updated.
+
+# Let's have $(bin_dir)/tools in front of the PATH so that we don't inavertedly
+# pick up the wrong binary somewhere. Watch out, $(shell echo $$PATH) will
+# still print the original PATH, since GNU make does not honor exported
+# variables: https://stackoverflow.com/questions/54726457
+export PATH := $(CURDIR)/$(bin_dir)/tools:$(PATH)
+
+CTR=docker
+
+TOOLS :=
+# https://github.com/helm/helm/releases
+TOOLS += helm=v3.13.3
+# https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl
+TOOLS += kubectl=v1.29.0
+# https://github.com/kubernetes-sigs/kind/releases
+TOOLS += kind=v0.20.0
+# https://www.vaultproject.io/downloads
+TOOLS += vault=1.15.4
+# https://github.com/Azure/azure-workload-identity/releases
+TOOLS += azwi=v1.2.0
```

Comment:
question: Where's azwi coming from? I can't see it as currently being a dependency of istio-csr? Same question for vault I think and maybe others.

Reply:
This library is part of the make-file modules, so it shouldn't be used in this project specifically.
That being said, it is also not used in approver-policy (the only other library that uses makefile-modules today), so we actually don't really need it.